integrasjonspunkt/src/test/java/no/difi/meldingsutveksling/noarkexchange/altinn/IntegrasjonspunktReceiveImplIntegrationTest.java

```diff
@@ -0,0 +1,75 @@
+package no.difi.meldingsutveksling.noarkexchange.altinn;
+
+import no.difi.meldingsutveksling.IntegrasjonspunktApplication;
+import no.difi.meldingsutveksling.dokumentpakking.xml.Payload;
+import no.difi.meldingsutveksling.domain.sbdh.EduDocument;
+import no.difi.meldingsutveksling.noarkexchange.*;
+import no.difi.meldingsutveksling.noarkexchange.receive.InternalQueue;
+import no.difi.meldingsutveksling.noarkexchange.schema.AppReceiptType;
+import no.difi.meldingsutveksling.noarkexchange.schema.PutMessageRequestType;
+import no.difi.meldingsutveksling.noarkexchange.schema.PutMessageResponseType;
+import no.difi.meldingsutveksling.noarkexchange.schema.receive.StandardBusinessDocument;
+import no.difi.vefa.peppol.lookup.api.LookupException;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.boot.test.WebIntegrationTest;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import javax.xml.bind.JAXBContext;
+import javax.xml.bind.JAXBElement;
+import javax.xml.bind.JAXBException;
+import javax.xml.bind.Unmarshaller;
+import javax.xml.transform.stream.StreamSource;
+import java.io.IOException;
+
+import static org.mockito.Mockito.*;
+
+/**
+ * Integration test for {@link IntegrajonspunktReceiveImpl} (via {@link InternalQueue}).
+ *
+ * Mock overrides are configured in {@link no.difi.meldingsutveksling.noarkexchange.IntegrasjonspunktIntegrationTestConfig}
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(IntegrasjonspunktApplication.class)
+@WebIntegrationTest
+@ActiveProfiles("test")
+public class IntegrasjonspunktReceiveImplIntegrationTest {
+
+    @Autowired
+    InternalQueue internalQueue;
+
+    @Autowired
+    IntegrajonspunktReceiveImpl integrajonspunktReceive;
+
+    @Test
+    public void receiveMessageTest() throws LookupException, JAXBException, IOException, MessageException {
+        AppReceiptType appReceiptTypeMock = mock(AppReceiptType.class);
```

Comment:
Her ville jeg trukket ut endel av oppsettet i private metoder sånn at det man tester blir mer synlig. Det er rett og slett for mye setup til å ha i test metoden